packages/headless/src/api/search/search/raw.ts

```diff
@@ -1,27 +1,21 @@
-type BaseRaw = {
+export interface Raw {
```

Comment:
So as we discussed offline, we do indeed include those fields by defaults at the moment for internal reason but the `raw` field is meant to be opaque and contain content derived from the fields to include.

So I would personally prefer to have those fields optional. They would still help with auto-completion but users would have to check for nullable.

cc: @dreisch-coveo

Reply:
to clarify a little, which fields are included in the raw depends on the request. by default we return all of a document's fields.

which fields might exist depend entirely on the source.

as far as I know only `urihash` is ABSOLUTELY guaranteed to exist. all the others are optional.

`permanentId` is populated on MOST sources by convention only.

for example using PushAPI and requesting `fieldsToInclude: []`, only `urihash` will be returned.

Reply:
I see, I could mark `permanentid` and `sourcetype` (as well as the removed keys) as optional, but my concern is what if someone indexes a different type into the field?

e.g. a `string[]` instead of a `string`, a `number` instead of a `string`.

Reply:
i do see utility in providing completion on "common" fields even if not guaranteed. if you want to support that I guess either assume the type matches what our sources usually do or just don't try to provide type information for them?

Reply:
Hmm, the more I think about it, the more I like the idea of keeping the property names optional, with their expected types. If the type is wrong, one would need to cast, similar to if the type was marked `unknown`.

The difference being with the expected type, this should work in 99% of cases.
In the 1% where it does not work, the error would be caught at run-time, instead of 100% at compile time with the `unknown` approach.

Reply:
If we don't use the type `unknown`, perhaps we should add a `@warning` of some sort.